Automattic/jetpack
# Nav Unification: Set default view with a direct server request

Fixes https://github.com/Automattic/jetpack/issues/21285

#### Changes proposed in this Pull Request:
Previously, when a user clicked on the button that switches from the classic view to the default view of a WP Admin screen, we triggered an AJAX request to store the newly preferred screen before the `href` attribute of the button actually redirected the user to the default view.

However, some browsers like Firefox interpret that AJAX request as a tracking intent because the domain of the AJAX request doesn't match the domain of the `href` attribute, effectively blocking the AJAX request and causing the preferred view to never be updated.

As a workaround, this PR removes the AJAX request and replaces the `href` attribute with a normal server request to ensure the preferred view is updated before we redirect the user to the default view.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Make sure to use Firefox for testing.
- Prepare the testing environment:
  - Simple sites: Apply D69415-code to your WP.com sandbox, and sandbox both the API and a Simple site.
  - Atomic sites: Install Jetpack Beta and switch to the `fix/nav-unification-set-default-view` branch.
- Go to https://wordpress.com and switch to your testing site.
- Go to "Posts > All posts".
- Open the "Screen Options" tab and switch to the classic view.
- Make sure you're redirected to the WP Admin equivalent screen.
- Make sure that "Posts > All posts" points now to WP Admin.
- Open the "Screen Options" tab and switch to the default view.
- Make sure you're redirected to the Calypso equivalent screen.
- Make sure that "Posts > All posts" points now to Calypso.
